### DISCUSSION TITLE
Shaders on Linux

### DISCUSSION THREAD

**Discussion by MetalMaxMX** [MonoGame/MonoGame#8185] [Desktop computers]
Hey! I have a bit of a question. I read that in order to actually use shaders on MonoGame from a Linux setup, you need to compile them through an application called FXC (Effects compiler), which means you need to use Wine in order to get those shaders compiled and running. I investigated this on the older forums and most of the answers seem to point that this was indeed the case. What I'd like to know is that if this still holds true or there are now alternative ways to get shaders working on MonoGame without the need for either Wine or FXC? I ask because most of the replies there were quite old and I am not sure if this is still the case to this day. (The docs also seem to point this is the case, but again, I am not sure if they were updated for this either, that's why I am asking) 

Cheers!
MetalMaxMX.

**Comment by Apostolique** [comment]
I've been using this script: <https://learn-monogame.github.io/how-to/automate-release/>. This one is for Ubuntu but you can adapt for any distro.

**Reply by MetalMaxMX** [reply]
Hello! Thanks for answering! 

Looks like a very useful script, I took a glance at it and it seems like it does link Wine with FXC to build shaders. I was mostly wondering if it was possible to have something like, say, if you focus solely on the DekstopGL port then you can bring GLSL shaders instead of HLSL, and that way you are able to skip the Wine setup, but still, the fact that it also works as a way to publish releases for all three OSes is pretty cool! 

Cheers!
MetalMaxMX
